packages/commonwealth/server/routes/deletePoll.ts

```diff
@@ -0,0 +1,76 @@
+import { AppError, ServerError } from 'common-common/src/errors';
+import moment from 'moment';
+import { getNextPollEndingTime } from '../../shared/utils';
+import type { DB } from '../models';
+import { findOneRole } from '../util/roles';
+import type { ValidateChainParams } from '../middleware/validateChain';
+import type { TypedRequestBody, TypedResponse } from '../types';
+import { success } from '../types';
+import { Op } from 'sequelize';
+
+export const Errors = {
+  InvalidChainComm: 'Invalid chain or community',
+  NotLoggedIn: 'Not logged in',
+  NotAuthor: 'Not the Author or Admin',
+  NoThread: 'No thread provided',
+};
+
+type DeletePollReq = {
+  thread_id: number;
+  poll_id: number;
+  chain_id: string;
+} & ValidateChainParams;
+
+type DeletePollResp = {};
+
+const deletePoll = async (
+  models: DB,
+  req: TypedRequestBody<DeletePollReq>,
+  res: TypedResponse<DeletePollResp>
+) => {
+  if (!req.user) throw new AppError(Errors.NotLoggedIn);
+
+  const { thread_id, poll_id, chain_id } = req.body;
+
+  try {
+    const thread = await models.Thread.findOne({
+      where: {
+        id: thread_id,
+      },
+    });
+
+    if (!thread) throw new AppError(Errors.NoThread);
+    const userOwnedAddressIds = (await req.user.getAddresses())
```

Comment:
Don't we have a helper for all this admin logic?

Reply:
we have this exact pattern in a dozen + places for getting the user's addresses. i think it's fine for the moment.